src/Onboarding/Routes/CurrencyRoute.php

```diff
@@ -80,36 +80,22 @@ public function registerRoute()
                         'value' => [
                             'type' => 'string',
                             'required' => true,
-                            // 'validate_callback' => [ $this, 'validateSetting' ],
+                            'validate_callback' => [$this, 'validateSetting'],
                             'sanitize_callback' => 'sanitize_text_field',
                         ],
                     ],
                 ],
-                'schema' => [$this, 'getSchema'],
             ]
         );
     }
 
     /**
-     * @since 2.8.0
-     * @return array
+     * Limits the symbol to a 2-letter country code
      *
+     * @unreleased
      */
-    public function getSchema()
+    public function validateSetting($value): bool
     {
-        return [
-            // This tells the spec of JSON Schema we are using which is draft 4.
-            '$schema' => 'http://json-schema.org/draft-04/schema#',
-            // The title property marks the identity of the resource.
-            'title' => 'onboarding',
-            'type' => 'object',
-            // In JSON Schema you can specify object properties in the properties attribute.
-            'properties' => [
-                'currencyCode' => [
-                    'description' => esc_html__('Two letter code representing a country.', 'give'),
-                    'type' => 'string',
-                ],
-            ],
-        ];
+        return preg_match('/^[A-Z]{2}$/i', $value) === 1;
```

Comment:
@jonwaldstein Solid idea. Why are you doing `json_decode`?

Reply:
> we can add give_get_currencies_list() as a property of the route class since it's used in handleRequest as well.

I'm not sure what you mean by this.

Reply:
I noticed the payload is not being decoded within the validate callback, not sure why.  

I just meant adding a `$currencyList` property to the class because it's also use in `handleRequest`